### PR TITLE
Make Mining Challenge component more modular

### PR DESCRIPTION
**Pull request by shaavan**
This PR is an effort in line to solve issue #448

According to [this](https://github.com/saving-satoshi/saving-satoshi/pull/527#pullrequestreview-1495974120) comment:

> By introducing all these kinds of variants we lose flexibility/composability. I was thinking we should do something more along these lines:
https://codesandbox.io/s/core-components-3tjy3n

This PR follows the example presented above to make the mining challenge component flexible and increase the compensability of components.

### CODE REVIEW

**Comment by shaavan** [comment]
cc @afterburn

File: content/lessons/chapter-2/mining-1.tsx [hunk 1]
```diff
@@ -187,53 +192,100 @@ export default function Mining1({ lang }) {
...
+              >
+                {Intl.NumberFormat().format(transactionsConfirmed)}
+              </div>
+              <TitleCard
+                title={t('chapter_two.mining_one.progress_bar_three')}
+                disabled={transactionsConfirmed === 0}
+                size={'small'}
+              />
+            </Card>
+            <Card className="w-max px-10 py-2.5">
```
[the remark this answers is not in the document]
Comment (benalleng) [reply content/lessons/chapter-2/mining-1.tsx:270]: same width issue here

File: content/lessons/chapter-2/mining-1.tsx [hunk 2]
```diff
@@ -187,53 +192,100 @@ export default function Mining1({ lang }) {
...
+              <div
+                className={clsx('text-2xl font-normal', {
```
Comment (benalleng) [inline content/lessons/chapter-2/mining-1.tsx:272]: this needs to be `font-space-mono`

File: content/lessons/chapter-2/mining-1.tsx [hunk 3]
```diff
@@ -187,53 +192,100 @@ export default function Mining1({ lang }) {
...
       <div className="grid grid-cols-1 justify-center justify-items-center md:my-auto md:flex md:flex-row">
-        <div className="fade-in grid w-full grid-cols-1 items-center px-[15px] py-[25px] md:order-last md:my-0 md:mx-[30px] md:w-[405px] md:p-[25px]">
-          <div
-            className={clsx(
-              'relative mb-2.5 font-nunito text-lg font-semibold',
-              {
-                'fade-in text-white': blocks !== 0,
-                'text-black/50': blocks === 0,
+        <div className="fade-in grid w-full grid-cols-1 items-center px-[15px] py-[25px] md:order-last md:mx-[30px] md:my-0 md:w-[405px] md:p-[25px]">
+          <Card className="font-nunito text-lg font-semibold" transparent>
```
Comment (benalleng) [inline content/lessons/chapter-2/mining-1.tsx:196]: you will want to add a `mt-0` on this Card to cancel the top-margin for this first item

File: ui/common/ProgressBar.tsx
Comment (benalleng) [inline ui/common/ProgressBar.tsx:1]: With us ultimately moving the progress down to 100 i don't think its necessary to do the full transform math instead lets use the simplified version 
```
style={{
    transform: `translate3d(-${total - progress}%,0,0)`,
}}
```

**Review by benalleng — CHANGES_REQUESTED** [review]
mostly styling changes but it all looks good! If we can get some approval of this PR i'll go ahead and close my PR with the other approach

**Comment by shaavan** [comment]
Updated from e4faaa3216fd79d2731ed0a0068f1d9e8895cd1b to f2792a38f471fb6a75da4b2dca348ada7cac9aab ([pr535.01](https://github.com/shaavan/saving-satoshi/commits/pr535.01) -> [pr535.02](https://github.com/shaavan/saving-satoshi/commits/pr535.02), [diff](https://github.com/shaavan/saving-satoshi/compare/pr535.01..pr535.02))
Addressed @benalleng comments

**Updates:**
1. Added highlight attributes to Card as mentioned in the above [comments](https://github.com/saving-satoshi/saving-satoshi/pull/535#discussion_r1244165238).
2. Made appropriate font change.
3. Introduce the new `firstItem` prop to Card to set the appropriate margin for the first element.

@benalleng

> ..the width needs some adjustment as it is not ideal in mobile format

I tried noticing the difference between **Master Branch** and the **PR** on my iPhone 13, but except for a slight difference in the gap between the final two Cards, I couldn't observe any other change.

**Screenshot:**

|Master|PR|
|-------|---|
|IMG_3905|IMG_3904|

Are you pointing towards this difference in your comment, or is there some major difference noticeable on another device?

File: content/lessons/chapter-2/mining-1.tsx [hunk 4]
```diff
@@ -187,53 +192,100 @@ export default function Mining1({ lang }) {
...
+                setHashPower={setHashPower}
+                onButtonClick={turnOnButton}
+                step={step}
+              ></StartButton>
+            </div>
+            <ProgressBar progress={hashPower / 440} variant={'bars'} />{' '}
+            {/* progress = hashPower * 100/maxHashPower */}
+          </Card>
+          <div className="flex items-center justify-between gap-x-2.5 text-center font-space-mono">
+            <Card className="w-max px-3.5 py-2.5">
```
[the remark this answers is not in the document]
Comment (shaavan) [reply content/lessons/chapter-2/mining-1.tsx]: Addressed the highlight in f2792a38f471fb6a75da4b2dca348ada7cac9aab

A question related to the width issue that I have mentioned in this [comment](https://github.com/saving-satoshi/saving-satoshi/pull/535#issuecomment-1611343875).

File: content/lessons/chapter-2/mining-1.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (shaavan) [reply content/lessons/chapter-2/mining-1.tsx]: Addressed the highlight in https://github.com/saving-satoshi/saving-satoshi/commit/f2792a38f471fb6a75da4b2dca348ada7cac9aab

A question related to the width issue that I have mentioned in this https://github.com/saving-satoshi/saving-satoshi/pull/535#issuecomment-1611343875.

File: content/lessons/chapter-2/mining-1.tsx (hunk 2 above)
Comment (shaavan) [reply content/lessons/chapter-2/mining-1.tsx]: Addressed the highlight in https://github.com/saving-satoshi/saving-satoshi/commit/f2792a38f471fb6a75da4b2dca348ada7cac9aab

Thanks for catching it!

File: ui/common/ProgressBar.tsx
Comment (shaavan) [reply ui/common/ProgressBar.tsx:1]: I agree with the change.

Addressed in b2fff5f90ad357681353f85f86442cf57c3d2d3b.
Thanks for the suggestion

**Comment by shaavan** [comment]
Updated from f2792a38f471fb6a75da4b2dca348ada7cac9aab to 5e050bb126dbc4f3b1a3c043363874a35404b4c0 ([pr535.02](https://github.com/shaavan/saving-satoshi/commits/pr535.02) -> [pr535.03](https://github.com/shaavan/saving-satoshi/commits/pr535.03), [diff](https://github.com/shaavan/saving-satoshi/compare/pr535.02..pr535.03))
Addressed @benalleng comment

Update:

1. Fix the sizing issue of "dual" Cards.

Screenshot:

**pr535.03:**

Screenshot 2023-06-29 at 4 50 28 PM
Screenshot 2023-06-29 at 4 50 40 PM

**Review by benalleng — APPROVED** [review]
LGTM! I'm satisfied with the reusability of the components and believe that Card.tsx can be used to quickly complete the Profile component in #523

**Pull request merged**